cmd/planner-api/run.go

```diff
@@ -126,11 +132,14 @@ var runCmd = &cobra.Command{
 			}
 			metricsServer := apiserver.NewMetricServer("0.0.0.0:8080", listener)
 			if err := metricsServer.Run(ctx); err != nil {
-				zap.S().Named("metrics_server").Fatalw("failed to run metrics server", "error", err)
+				zap.S().Named("metrics_server").Errorw("failed to run metrics server", "error", err)
 			}
+			wg.Done()
```

Comment:
ok but this is risky.  Recommended pattern is with `defer`. If in the future, we add a new exit path this will not work